### DISCUSSION TITLE
How to save in Dialogic 2 with code?

### DISCUSSION THREAD

**Discussion by creativecorvus3** [dialogic-godot/dialogic#1393] [Q&A] [answered]
In Dialogic 1 I would simply need to run this code _**Dialogic.save('saveslot_name')**_

For Dialogic 2 it changed it to be _**Dialogic.Save.save('saveslot_name')**_ yet I get this error _**Attempt to call function 'store_var' in base 'null instance' on a null instance**_ what do I need to do to fix this?

**Comment by Jowan-Spooner** [comment, marked as answer]
Hi this seems to be a bug. For whatever reason it fails to access the save folder directory. Can you give more info on your system?

**Reply by creativecorvus3** [reply]
![Screenshot 2023-01-02 115052](https://user-images.githubusercontent.com/114537351/210269776-0d73c38b-ed36-436f-b7a2-564e1a08e358.png)
![Screenshot 2023-01-02 115115](https://user-images.githubusercontent.com/114537351/210269777-57f6256c-1b1b-43b0-bb83-f75a3f9c0c99.png)
![Screenshot 2023-01-02 115158](https://user-images.githubusercontent.com/114537351/210269779-59ed3ac6-a165-4572-bf20-6766aaa6d64e.png)
![Screenshot 2023-01-02 115413](https://user-images.githubusercontent.com/114537351/210269781-ee1771c1-b09c-4803-ac62-2bce0f3cefa3.png)
The first three are from the debugger while the last image is my code itself. It should work as I am saving a string variable, but it comes with this error.

How I save my player game is whenever they create a character it takes that name VAR and adds a RAND generated number (in case they create a new character with the same name) and uses that to be the save file name.

**Reply by Jowan-Spooner** [reply]
This looks okay. I was rather wondering about: what alpha version, what godot beta version, what OS are you using. It seems like you are not using the current dialogic main branch (potentially the latest alpha release). I've pushed fixes/changes to the save system recently. Maybe you could try with the current WIP version?

**Reply by creativecorvus3** [reply]
Oh, sorry didn't really understand what you meant. I am using Godot Engine v4 beta 10 and Dialogic 2 Alpha 5. My OS is Windows 11 Home.

**Reply by Jowan-Spooner** [reply]
Hi late answer, but I really think you should try out the main branch, as I merged changes/fixes to the save system a few days after the latest alpha 5 release. So it would be very useful to me to know if those changes might have fixed your issue.

**Reply by creativecorvus3** [reply]
It works!

Sorry for the late reply was doing college work. But, yes I have used the new main branch and it does work! Saving works completely fine, although I got three new errors that I just commented out to run my game.

New errors:
addons\dialogic\Events\Background\default_background.gd
addons\dialogic\Events\Text\subsystem_text.gd

![Error1](https://user-images.githubusercontent.com/114537351/216868697-6fe378c6-ba1d-4900-a7a9-4c1542d37817.png)
![Error2](https://user-images.githubusercontent.com/114537351/216868698-25829fce-d36f-4f64-95fa-1a5952c3bbb7.png)
![Error3](https://user-images.githubusercontent.com/114537351/216868699-831ad92a-7c9f-42b9-843e-1a6a0a205cd3.png)
